File: C/练习/code_13_2.c
#include <stdio.h>


void print_num(void) {
    for (int num = 100; num <= 999; num++) {
        int x = num / 100;
        int y = num % 100 / 10;
        int z = num % 100 % 10;
        if (x * x * x + y * y * y + z * z * z == x * 100 + y * 10 + z) {
            printf("%d 是水仙花数\n", num);
        }
    }
}

int main(void) {
    print_num();
    return 0;
}

File: C/Replicate_With_Python/code_13_2.py
def print_num():
    for num in range(100, 1000):
        x = num // 100
        y = num % 100 // 10
        z = num % 100 % 10
        if x ** 3 + y ** 3 + z ** 3 == x * 100 + y * 10 + z:
            print(f"{num} 是水仙花数")
    return


def main():
    print_num()

if __name__ == '__main__':
    main()